### PR TITLE
Add support for queue_classic 3.1 and it's scheduling

### PR DESCRIPTION
**Pull request by jipiboily**
This is a follow up from https://github.com/QueueClassic/queue_classic_admin/pull/37. I basically checked out the branch and I'm going to start from there. The original PR went stall and this is required now for us internally.

TODO;
- [x] review & address my own review of https://github.com/QueueClassic/queue_classic_admin/pull/37
- [x] add a migration to remove all traces of queue_classic-later (DOESN'T MAKE SENSE TO DO THAT HERE, SCRATCH THAT)
- [x] see what's missing and add it here
- [x] test
- [x] ?

/cc @smathieu @ukd1

### CODE REVIEW

**Comment by jipiboily** [comment]
@smathieu this would be ready for review plz

File: app/controllers/queue_classic_admin/queue_classic_jobs_controller.rb [hunk 1]
```diff
@@ -14,20 +15,22 @@ def destroy
     end
 
     def destroy_all
-      filter_jobs(QueueClassicJob).delete_all
+      filter_jobs(QueueClassicJob)
+      @queue_classic_jobs.ready.delete_all
```
Comment (smathieu) [inline app/controllers/queue_classic_admin/queue_classic_jobs_controller.rb]: Why is `@queue_classic_jobs.ready` duplicated everywhere?

File: app/controllers/queue_classic_admin/queue_classic_jobs_controller.rb [hunk 2]
```diff
@@ -14,20 +15,22 @@ def destroy
...
+      filter_jobs(QueueClassicJob)
+      @queue_classic_jobs.ready.where('locked_at < ?', 5.minutes.ago).update_all(locked_at: nil)
       redirect_to queue_classic_jobs_url
     end
 
     def bulk_custom_action
-      jobs = filter_jobs(QueueClassicJob)
-
+      filter_jobs(QueueClassicJob)
+      @queue_classic_jobs.ready
```
Comment (smathieu) [inline app/controllers/queue_classic_admin/queue_classic_jobs_controller.rb]: I don't think this does anything?

File: spec/support/queue_classic_helpers.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+module QueueClassicAdmin::QueueClassicSpecHelper
+  def create_job_qc_job(args = {})
```
Comment (smathieu) [inline spec/support/queue_classic_helpers.rb]: Thta's a strange method name

**Comment by smathieu** [comment]
On a more general note, I'm not sure if I' have built this as a separate controller. It seems that now that everything is in one table, it would make more sense to just have a filter for :all, :ready & :scheduled in the UI and have a single controller.

**Comment by jipiboily** [comment]
Yeah, I don't think 2 controllers is ideal. I took over the code and somewhat cleaned it. I had to call it a wrap and move on to something else. We can wait and modify this but I want have time before 10+ days. If you have time before the end of the week to clean even more, great. But I think we should at least use this in the main app, maybe not merge this branch and clean it later before merging?

File: spec/support/queue_classic_helpers.rb (hunk 3 above)
Comment (jipiboily) [inline spec/support/queue_classic_helpers.rb]: lol, yeah, modified the name, apparently I missed the double job.

File: app/controllers/queue_classic_admin/queue_classic_jobs_controller.rb (hunk 2 above)
Comment (jipiboily) [inline app/controllers/queue_classic_admin/queue_classic_jobs_controller.rb]: Indeed. It should be `@queue_classic_jobs = @queue_classic_jobs.ready`.

Comment (smathieu) [inline app/controllers/queue_classic_admin/queue_classic_jobs_controller.rb]: This should have been caught by a test

Comment (jipiboily) [inline app/controllers/queue_classic_admin/queue_classic_jobs_controller.rb]: Totally agree. Basically, this shouldn't be merged before more work, it's not future proof and super hacky.

**Comment by jipiboily** [comment]
Basically, this shouldn't be merged before more work, it's not future proof & not well tested. Also super hacky.

**Pull request merged**